File: medplum-java/medplum-java-model/src/main/java/com/fhir/Binary.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableBinary.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Binary {
  Optional<code> contentType();

  Optional<Reference> securityContext();

  Optional<base64Binary> data();

  Optional<id> id();

  String resourceType();

  Optional<Meta> meta();

  Optional<uri> implicitRules();

  Optional<code> language();

  static ImmutableBinary.ResourceTypeBuildStage builder() {
    return ImmutableBinary.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/Binary.ts
import { base64Binary } from "./base64Binary";
import { code } from "./code";
import { id } from "./id";
import { Meta } from "./Meta";
import { Reference } from "./Reference";
import { uri } from "./uri";

export interface Binary {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    contentType: code | null | undefined;
    securityContext: Reference | null | undefined;
    data: base64Binary | null | undefined;
}
